File: scratch/add_cache_busting.js
const fs = require('fs');
const path = require('path');

function processHtmlFile(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');

  // Replace CSS links
  content = content.replace(
    /href=["'](\.\.\/)?css\/(maincss|buyer)\.css(?:\?[^"']*)?["']/g,
    'href="$1css/$2.css?v=2"'
  );

  // Replace JS script sources
  content = content.replace(
    /src=["'](\.\.\/)?js\/(mainjs|buyer|ai-recommend)\.js(?:\?[^"']*)?["']/g,
    'src="$1js/$2.js?v=2"'
  );

  fs.writeFileSync(filePath, content, 'utf8');
  console.log(`Updated cache-busting in ${filePath}`);
}

function walk(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.git' && file !== '.gemini' && file !== 'knowledge') {
        walk(fullPath);
      }
    } else if (file.endsWith('.html')) {
      processHtmlFile(fullPath);
    }
  }
}

walk('d:\\FinalWorkWeb');

File: scratch/add_cache_busting.py
import os
import re

def process_html_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Pattern for css files (maincss.css and buyer.css)
    # Match href="../css/maincss.css" or href="../css/buyer.css" possibly with existing version
    content = re.sub(
        r'href=["\'](\.\./)?css/(maincss|buyer)\.css(?:\?[^"\']*)?["\']',
        r'href="\1css/\2.css?v=2"',
        content
    )

    # Pattern for js files (mainjs.js, buyer.js, ai-recommend.js)
    content = re.sub(
        r'src=["\'](\.\./)?js/(mainjs|buyer|ai-recommend)\.js(?:\?[^"\']*)?["\']',
        r'src="\1js/\2.js?v=2"',
        content
    )

    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    print(f"Updated cache-busting in {file_path}")

def main():
    root_dir = r"d:\FinalWorkWeb"
    for root, dirs, files in os.walk(root_dir):
        # Skip node_modules or .git
        if 'node_modules' in root or '.git' in root or '.gemini' in root:
            continue
        for file in files:
            if file.endswith('.html'):
                process_html_file(os.path.join(root, file))

if __name__ == '__main__':
    main()
